diff --git a/core/include/physics/MeshCollider.hpp b/core/include/physics/MeshCollider.hpp
--- a/core/include/physics/MeshCollider.hpp
+++ b/core/include/physics/MeshCollider.hpp
@@ -1,43 +1,45 @@
diff --git a/core/src/physics/MeshCollider.cpp b/core/src/physics/MeshCollider.cpp
--- a/core/src/physics/MeshCollider.cpp
+++ b/core/src/physics/MeshCollider.cpp
@@ -1,102 +1,196 @@
diff --git a/tests/physics/test_mesh_collider.cpp b/tests/physics/test_mesh_collider.cpp
--- a/tests/physics/test_mesh_collider.cpp
+++ b/tests/physics/test_mesh_collider.cpp
@@ -1,162 +1,224 @@